[BUG]: Cannot create Blobstores with new provider version v1.3.0

### Terraform Version

v1.13.0

### Provider Version

1.3.0

### Sonatype Nexus Repository Manager version

3.90.1

### High Availability enabled?

Yes - HA is enabled

### Describe the Issue or Bug

Hi, 
I cannot create blob-stores with provider version v1.3.0. 
The same piece of code used to work with provider version 0.14.



### Terraform Configuration Files

```hcl
My Terraform snippet:


resource "sonatyperepo_blob_store_s3" "maven_hazelcast" {
  name = "s3-maven-hazelcast"
  bucket_configuration = {
    bucket = {
      name   = "eun1-lab-binary-blob-store-s3-maven"
      prefix = "hazelcast"
      region = "eu-north-1"
    }
    bucket_security = {
      access_key_id     = var.NEXUS_S3_ACCESS_KEY
      secret_access_key = var.NEXUS_S3_SECRET_KEY
    }
    encryption = {
      encryption_key  = var.kms_key_arn
      encryption_type = "kmsManagedEncryption"
    }
  }

}
```

### Steps to Reproduce

1. Use added code snippet and run terraform plan (This is succesful)
2. Run terraform apply (it fails)

### Logs & Output

```shell
Output of Terraform plan:


  # sonatyperepo_blob_store_s3.maven_hazelcast will be created
  + resource "sonatyperepo_blob_store_s3" "maven_hazelcast" {
      + bucket_configuration = {
          + bucket                 = {
              + name   = "eun1-lab-binary-blob-store-s3-maven"
              + prefix = "hazelcast"
              + region = "eu-north-1"
            }
          + bucket_security        = {
              + access_key_id     = (sensitive value)
              + secret_access_key = (sensitive value)
            }
          + encryption             = {
              + encryption_key  = (sensitive value)
              + encryption_type = "kmsManagedEncryption"
            }
          + pre_signed_url_enabled = false
        }
      + last_updated         = (known after apply)
      + name                 = "s3-maven-hazelcast"
      + type                 = "s3"
    }

Plan: 1 to add, 0 to change, 0 to destroy.


Output of Terraform Apply:


sonatyperepo_blob_store_s3.maven_hazelcast: Creating...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-sonatyperepo_v1.3.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10549d268]

goroutine 97 [running]:
terraform-provider-sonatyperepo/internal/provider/model.(*BlobStoreS3Encryption).MapToApi(0x140003941b0, 0x0)
	terraform-provider-sonatyperepo/internal/provider/model/blob_store.go:269 +0x38
terraform-provider-sonatyperepo/internal/provider/model.(*BlobStoreS3BucketConfigurationModelV1).MapToApi(0x14000373260, 0x14000524ed0)
	terraform-provider-sonatyperepo/internal/provider/model/blob_store.go:216 +0x48
terraform-provider-sonatyperepo/internal/provider/model.(*BlobStoreS3ModelV1).MapToApi(0x140004e0240, 0x14000524ed0)
	terraform-provider-sonatyperepo/internal/provider/model/blob_store.go:175 +0x84
terraform-provider-sonatyperepo/internal/provider/blob_store.(*blobStoreS3Resource).Create(0x140004b2500, {0x105d03138, 0x14000170db0}, {{{{0x105d087a8, 0x140003d0f90}, {0x105b81740, 0x140003a4d20}}, {0x105d0a620, 0x140004be2d0}}, {{{0x105d087a8, ...}, ...}, ...}, ...}, ...)
	terraform-provider-sonatyperepo/internal/provider/blob_store/blob_store_s3_resource.go:159 +0x1b8
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x14000186a08, {0x105d03138, 0x14000170db0}, 0x140005254c0, 0x14000525490)
	github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/fwserver/server_createresource.go:127 +0x644
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x14000186a08, {0x105d03138, 0x14000170db0}, 0x140003724d0, 0x140005255e8)
	github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/fwserver/server_applyresourcechange.go:63 +0x434
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x14000186a08, {0x105d03138?, 0x14000170cc0?}, 0x14000172410)
	github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/proto6server/server_applyresourcechange.go:71 +0x430
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000222c80, {0x105d03138?, 0x140001701e0?}, 0x14000176080)
	github.com/hashicorp/terraform-plugin-go@v0.27.0/tfprotov6/tf6server/server.go:989 +0x294
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x105cc0b20, 0x14000222c80}, {0x105d03138, 0x140001701e0}, 0x14000176000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.27.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:687 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400020c200, {0x105d03138, 0x14000170150}, 0x140004e00c0, 0x140002eafc0, 0x10658f928, 0x0)
	google.golang.org/grpc@v1.73.0/server.go:1405 +0xc9c
google.golang.org/grpc.(*Server).handleStream(0x1400020c200, {0x105d03c10, 0x1400037e000}, 0x140004e00c0)
	google.golang.org/grpc@v1.73.0/server.go:1815 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.73.0/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
	google.golang.org/grpc@v1.73.0/server.go:1046 +0x138

Error: The terraform-provider-sonatyperepo_v1.3.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/internal/provider/model/blob_store.go
+++ b/internal/provider/model/blob_store.go
@@ -266,6 +266,9 @@ func (m *BlobStoreS3Encryption) MapFromApi(api *v3.S3BlobStoreApiEncryption) {
 }
 
 func (m *BlobStoreS3Encryption) MapToApi(api *v3.S3BlobStoreApiEncryption) {
+	if api == nil {
+		api = v3.NewS3BlobStoreApiEncryptionWithDefaults()
+	}
 	api.EncryptionType = m.EncryptionType.ValueStringPointer()
 	api.EncryptionKey = m.EncryptionKey.ValueStringPointer()
 }

